GremlinServerPlugin not working

Since Release 2026.3.2, the `GremlinServerPlugin` is no longer working. When starting the `server.sh` with to following option:

```
-Darcadedb.server.plugins=GremlinServer:com.arcadedb.server.gremlin.GremlinServerPlugin
```

the following error occur in the service logs:

```
 2026-03-25 14:42:50.475 INFO  [ServerGremlinExecutor] Initialized GremlinExecutor and preparing GremlinScriptEngines instances.Script compilation FAILED 1+1 took 357ms {}
 org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
 Script1.groovy: -1: The name valueOf is already declared @ line -1, column -1.
 1 error
 
 	at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:292)
 	at org.codehaus.groovy.control.CompilationUnit$IPrimaryClassNodeOperation.doPhaseOperation(CompilationUnit.java:964)
 	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:668)
 	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:642)
 	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:399)
 	at groovy.lang.GroovyClassLoader.lambda$parseClass$5(GroovyClassLoader.java:345)
 	at org.codehaus.groovy.runtime.memoize.ConcurrentCommonCache.getAndPut(ConcurrentCommonCache.java:138)
 	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:345)
 	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:329)
 	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:303)
 	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.lambda$load$0(GremlinGroovyScriptEngine.java:824)
 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
 	at java.base/java.util.concurrent.CompletableFuture.asyncSupplyStage(CompletableFuture.java:1782)
 	at java.base/java.util.concurrent.CompletableFuture.supplyAsync(CompletableFuture.java:2005)
 	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.load(GremlinGroovyScriptEngine.java:822)
 	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.load(GremlinGroovyScriptEngine.java:817)
 	at com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.lambda$new$0(BoundedLocalCache.java:3117)
 	at com.github.benmanes.caffeine.cache.LocalCache.lambda$statsAware$0(LocalCache.java:144)
 	at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$16(BoundedLocalCache.java:1968)
 	at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1916)
 	at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:1966)
 	at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:1949)
 	at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:113)
 	at com.github.benmanes.caffeine.cache.LocalLoadingCache.get(LocalLoadingCache.java:67)
 	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.getScriptClass(GremlinGroovyScriptEngine.java:572)
 	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:379)
 	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:231)
 	at org.apache.tinkerpop.gremlin.groovy.engine.GremlinExecutor.lambda$eval$0(GremlinExecutor.java:314)
 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
 	at java.base/java.lang.Thread.run(Thread.java:1583)
Could not initialize gremlin-groovy GremlinScriptEngine as init script could not be evaluated
java.util.concurrent.CompletionException: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
Script1.groovy: -1: The name valueOf is already declared @ line -1, column -1.
1 error

	at java.base/java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:413)
	at java.base/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2118)
	at org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor.lambda$new$4(ServerGremlinExecutor.java:170)
	at java.base/java.util.LinkedHashMap$LinkedKeySet.forEach(LinkedHashMap.java:729)
	at org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor.<init>(ServerGremlinExecutor.java:162)
	at org.apache.tinkerpop.gremlin.server.GremlinServer.<init>(GremlinServer.java:130)
	at org.apache.tinkerpop.gremlin.server.GremlinServer.<init>(GremlinServer.java:93)
	at com.arcadedb.server.gremlin.GremlinServerPlugin.startService(GremlinServerPlugin.java:97)
	at com.arcadedb.server.plugin.PluginManager.startPlugins(PluginManager.java:203)
	at com.arcadedb.server.ArcadeDBServer.start(ArcadeDBServer.java:194)
	at com.arcadedb.server.ArcadeDBServer.main(ArcadeDBServer.java:126)
Caused by: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
Script1.groovy: -1: The name valueOf is already declared @ line -1, column -1.
1 error

	at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:292)
	at org.codehaus.groovy.control.CompilationUnit$IPrimaryClassNodeOperation.doPhaseOperation(CompilationUnit.java:964)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:668)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:642)
	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:399)
	at groovy.lang.GroovyClassLoader.lambda$parseClass$5(GroovyClassLoader.java:345)
	at org.codehaus.groovy.runtime.memoize.ConcurrentCommonCache.getAndPut(ConcurrentCommonCache.java:138)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:345)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:329)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:303)
	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.lambda$load$0(GremlinGroovyScriptEngine.java:824)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture.asyncSupplyStage(CompletableFuture.java:1782)
	at java.base/java.util.concurrent.CompletableFuture.supplyAsync(CompletableFuture.java:2005)
	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.load(GremlinGroovyScriptEngine.java:822)
	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.load(GremlinGroovyScriptEngine.java:817)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.lambda$new$0(BoundedLocalCache.java:3117)
	at com.github.benmanes.caffeine.cache.LocalCache.lambda$statsAware$0(LocalCache.java:144)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$16(BoundedLocalCache.java:1968)
	at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1916)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:1966)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:1949)
	at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:113)
	at com.github.benmanes.caffeine.cache.LocalLoadingCache.get(LocalLoadingCache.java:67)
	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.getScriptClass(GremlinGroovyScriptEngine.java:572)
	at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:379)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:231)
	at org.apache.tinkerpop.gremlin.groovy.engine.GremlinExecutor.lambda$eval$0(GremlinExecutor.java:314)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)

```

Connecting to the service via WebSocket is not possible.

--- a/gremlin/pom.xml
+++ b/gremlin/pom.xml
@@ -41,7 +41,7 @@
         <commons-configuration2.version>2.13.0</commons-configuration2.version>
         <commons-beanutils.version>1.11.0</commons-beanutils.version>
         <commons-lang3.version>3.20.0</commons-lang3.version>
-        <groovy.version>5.0.4</groovy.version>
+        <groovy.version>4.0.28</groovy.version>
         <antlr4.version>4.9.1</antlr4.version>
     </properties>
 

--- a/gremlin/src/test/java/com/arcadedb/gremlin/GremlinGroovyEngineTest.java
+++ b/gremlin/src/test/java/com/arcadedb/gremlin/GremlinGroovyEngineTest.java
@@ -1,0 +1,53 @@
+/*
+ * Copyright © 2021-present Arcade Data Ltd (info@arcadedata.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-FileCopyrightText: 2021-present Arcade Data Ltd (info@arcadedata.com)
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.arcadedb.gremlin;
+
+import org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine;
+import org.apache.tinkerpop.gremlin.jsr223.CoreGremlinPlugin;
+import org.apache.tinkerpop.gremlin.jsr223.Customizer;
+import org.junit.jupiter.api.Test;
+
+import javax.script.ScriptException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Verifies that the Gremlin Groovy script engine initialises correctly with the TinkerPop
+ * core imports.  With Groovy 5 (incompatible with TinkerPop 3.8.x) this fails with
+ * "The name valueOf is already declared" because {@code ImportGroovyCustomizer} adds a
+ * wildcard static import ({@code import static Enum.*}) for every enum in
+ * {@link org.apache.tinkerpop.gremlin.jsr223.CoreImports}, and Groovy 5 rejects duplicate
+ * {@code valueOf} symbols introduced by multiple such imports.
+ *
+ * <p>Regression test for <a href="https://github.com/ArcadeData/arcadedb/issues/3724">#3724</a>.
+ */
+class GremlinGroovyEngineTest {
+
+  @Test
+  void groovyEngineInitializesWithCoreImports() throws ScriptException {
+    final Customizer[] customizers = CoreGremlinPlugin.instance()
+        .getCustomizers("gremlin-groovy")
+        .orElse(new Customizer[0]);
+
+    final GremlinGroovyScriptEngine engine = new GremlinGroovyScriptEngine(customizers);
+
+    // A successful eval proves the engine started without "valueOf is already declared"
+    assertThat(engine.eval("1 + 1")).isEqualTo(2);
+  }
+}

